lib/Widgets/SettingsPage/AbstractSettingsPage.vala

```diff
@@ -21,15 +21,59 @@
  * AbstractSettingsPage is a #Gtk.ScrolledWindow subclass with properties used
  * by other Granite settings widgets.
  */
-
 public abstract class Granite.SettingsPage : Gtk.ScrolledWindow {
+    protected string _icon_name;
+    protected string _title;
+
+    public enum StatusType {
+        ERROR,
+        OFFLINE,
+        SUCCESS,
+        WARNING,
+        NONE
+    }
+
+    /**
+     * Selects a colored icon to be displayed in a Granite.SettingsSidebar
+     */
+    public StatusType status_type { get; set; default = StatusType.NONE; }
+
+    /**
+     * A widget to display in place of an icon in a Granite.SettingsSidebar
+     */
+    public Gtk.Widget? display_widget { get; construct; }
+
+    /**
+     * A header to be sorted under in a Granite.SettingsSidebar
+     */
+    public string? header { get; construct; }
+
+    /**
+     * An status string associated with #this
+     */
+    public string status { get; set; }
```

Comment:
This `status` property isn't exactly clear on what it does. Perhaps say it is the description of the status like "enabled" and "disabled"?